doc/reference/reports.rst

```diff
@@ -176,9 +173,20 @@ For example, let's look at the Sale Order report from the Sale module::
     </template>
 
 
-The main template calls translate_doc with ``partner_id.lang`` as a parameter,
-which means it uses :ref:`a custom report model
-<reference/reports/custom_reports>` to access a ``res.partner`` record.
+The main template calls translate_doc with ``doc.partner_id.lang`` as a parameter, 
+which is the language of the partner_id of the document. This way, each Sale Order 
+will be printed in the language of the customer. If you wish to translate only the 
+body of the document, but keep the header and footer in a default language, you 
+could call the report's external layout this way::
+
+    <t t-call="report.external_layout">
+
+.. warning::
+
+    Please take note that this works only when calling external templates, you will not be
+    able to translate part of a document by setting a t-lang attribute on an xml node other 
+    than t-call.
```

Comment:
@smack815 well the goal here is to make the translate_doc 'hack' (because it's really ugly if you dive in the code) deprecated; so if we merge this in 8.0 we will make it the only documented way to translate a template (and the functionality is the same, so having 2 ways of doing exactly the same thing is unnecessarily complicated)